File: src/Topping.ts
export interface Lettuceable {
  lettuce: Topping;
}

export interface Tomatoeable {
  tomato: Topping;
}

export interface Spreadable {
  spread: Topping;
}

export interface Chilliable {
  chillies: Topping;
}

export interface GrilledOnionable {
  grilledOnion: Topping;
}

export enum Topping {
  Regular = "Regular",
  Light = "Light",
  ExtraLight = "ExtraLight",
  Well = "Well",
  ExtraWell = "ExtraWell"
}

export interface Onionable {
  onion: Topping;
}

export interface WholeGrilledOnionable {
  wholeGrilledOnion: Topping;
}

export interface RawOnionable {
  rawOnion: Topping;
}

export interface RawChoppedOnionable {
  rawChoppedOnion: Topping;
}

export interface ChoppedOnionable {
  choppedOnion: Topping;
}

File: src/topping.rs
pub struct Lettuceable {
    lettuce: Topping,
}

pub struct Tomatoeable {
    tomato: Topping,
}

pub struct Spreadable {
    spread: Topping,
}

pub struct Chilliable {
    chillies: Topping,
}

pub struct GrilledOnionable {
    grilled_onion: Topping,
}

pub enum Topping {
    Regular,
    Light,
    ExtraLight,
    Well,
    ExtraWell,
}

pub struct Onionable {
    onion: Topping,
}

pub struct WholeGrilledOnionable {
    whole_grilled_onion: Topping,
}

pub struct RawOnionable {
    raw_onion: Topping,
}

pub struct RawChoppedOnionable {
    raw_chopped_onion: Topping,
}

pub struct ChoppedOnionable {
    chopped_onion: Topping,
}
